### PR TITLE
postgresqlPackages.timescaledb: 2.14.2 -> 2.17.2; adopt, nixfmt; postgresqlPackages.timescaledb_toolkit: 1.18.0 -> 1.19.0

### PR DESCRIPTION
**Pull request by kirillrdy**
Changes:
-  https://github.com/timescale/timescaledb/releases/tag/2.17.2
-  https://github.com/timescale/timescaledb/releases/tag/2.17.1
 - https://github.com/timescale/timescaledb/releases/tag/2.17.0
 - https://github.com/timescale/timescaledb/releases/tag/2.16.1
 - https://github.com/timescale/timescaledb/releases/tag/2.16.0
 - https://github.com/timescale/timescaledb/releases/tag/2.15.3
 - https://github.com/timescale/timescaledb/releases/tag/2.15.2
 - https://github.com/timescale/timescaledb/releases/tag/2.15.1
 - https://github.com/timescale/timescaledb/releases/tag/2.15.0
 - https://github.com/timescale/timescaledb-toolkit/releases/tag/1.19.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by wolfgangwalther — CHANGES_REQUESTED** [review]
Could you squash the two update commits + the "only supports postgresql 14" together in one commit and keep the maintainer + nixfmt commits separate?

**Comment by kirillrdy** [comment]
@wolfgangwalther going to draft until after 24.11 branch off

File: pkgs/servers/sql/postgresql/ext/timescaledb.nix [hunk 1]
```diff
@@ -32,16 +51,18 @@ stdenv.mkDerivation rec {
...
     changelog = "https://github.com/timescale/timescaledb/blob/${version}/CHANGELOG.md";
-    maintainers = [ ];
+    maintainers = [ maintainers.kirillrdy ];
     platforms = postgresql.meta.platforms;
     license = with licenses; if enableUnfree then tsl else asl20;
-    broken = versionOlder postgresql.version "13" ||
+    broken = versionOlder postgresql.version "14" ||
       # timescaledb supports PostgreSQL 17 from 2.17.0 on:
       # https://github.com/timescale/timescaledb/releases/tag/2.17.0
       # We can't upgrade to it, yet, because this would imply dropping support for
```
[the remark this answers is not in the document]
Comment (kirillrdy) [reply pkgs/servers/sql/postgresql/ext/timescaledb.nix]: done

File: pkgs/servers/sql/postgresql/ext/timescaledb.nix [hunk 2]
```diff
@@ -1,20 +1,39 @@
...
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  postgresql,
+  openssl,
+  libkrb5,
+  nixosTests,
+  enableUnfree ? true,
```
[the remark this answers is not in the document]
Comment (kirillrdy) [reply pkgs/servers/sql/postgresql/ext/timescaledb.nix]: done

**Comment by kirillrdy** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 348223`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>postgresql13JitPackages.timescaledb</li>
    <li>postgresql13JitPackages.timescaledb-apache</li>
    <li>postgresql13Packages.timescaledb</li>
    <li>postgresql13Packages.timescaledb-apache</li>
  </ul>
</details>
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>postgresql14JitPackages.timescaledb</li>
    <li>postgresql14JitPackages.timescaledb-apache</li>
    <li>postgresql14Packages.timescaledb</li>
    <li>postgresql14Packages.timescaledb-apache</li>
    <li>postgresql15JitPackages.timescaledb</li>
    <li>postgresql15JitPackages.timescaledb-apache</li>
    <li>postgresql15Packages.timescaledb</li>
    <li>postgresql15Packages.timescaledb-apache</li>
    <li>postgresql16JitPackages.timescaledb</li>
    <li>postgresql16JitPackages.timescaledb-apache</li>
    <li>postgresql16Packages.timescaledb</li>
    <li>postgresql16Packages.timescaledb-apache</li>
    <li>postgresql17JitPackages.timescaledb</li>
    <li>postgresql17JitPackages.timescaledb-apache</li>
    <li>postgresql17Packages.timescaledb</li>
    <li>postgresql17Packages.timescaledb-apache</li>
  </ul>
</details>

**Review by wolfgangwalther — CHANGES_REQUESTED** [review]
Could you squash the last commit "mention timescaledb" into the second commit "2.14.2 -> 2.17.1"?

Those changes really belong together, I think.

**Comment by kirillrdy** [comment]
> > There is now also an update to `timescaledb_toolkit` available, which adds support for PG17. I suggest you bump that here, too.
> 
> Did you have a look at that, too?
> 
> Without bumping timescaledb_toolkit we can't test `nixosTests.postgresql.timescaledb`, I think.

I can have a look into `timescaledb_toolkit`, `nixosTests.postgresql.timescaledb` builds, but master is still on 16.4

**Comment by philon123** [comment]
@kirillrdy you're awesome, thanks! Can I help with this somehow? I could maybe confirm it working on x86_64 with a real world use case. Not sure if I can contribute to the actual PR though.

Would it be fine to update to the newest version, v2.17.2?

Any idea if this will make NixOS 24.11?

**Comment by kirillrdy** [comment]
> @kirillrdy you're awesome, thanks! Can I help with this somehow? I could maybe confirm it working on x86_64 with a real world use case. Not sure if I can contribute to the actual PR though.
> 
> Would it be fine to update to the newest version, v2.17.2?
> 
> Any idea if this will make NixOS 24.11?

any testing is much appreciated, I only looked at whats in `nixosTests.timescaledb`

yes I will bump to  v2.17.2

no, this will not make to NixOS 24.11 due to breaking changes, but it doesn't mean it can not be installed on nixos 24.11

**Review by Ma27 — APPROVED** [review]

**Comment by kirillrdy** [comment]
> `postgresql17Packages.timescaledb_toolkit` doesn't build for me with this PR:
> 
> ```
>        > error: unexpected argument '--pg17' found
>        >
>        >   tip: a similar argument exists: '--pg16'
> ```

yes, looks like we need to bump cargo-pgrx too

File: pkgs/development/tools/rust/cargo-pgrx/default.nix [hunk 3]
```diff
@@ -77,4 +77,11 @@ in
     hash = "sha256-0m9oaqjU42RYyttkTihADDrRMjr2WoK/8sInZALeHws=";
     cargoHash = "sha256-9XTIcpoCnROP63ZTDgMMMmj0kPggiTazKlKQfCgXKzk=";
   };
+
+  cargo-pgrx_0_12_6 = generic {
```
Comment (Ma27) [inline pkgs/development/tools/rust/cargo-pgrx/default.nix:81]: Can't we bump the alpha version above?

**Review by Ma27 — COMMENTED** [review]
Ah right, forgot to check the v17 variant of the toolkit. Good thing I got distracted from merging I guess ;-)

File: pkgs/development/tools/rust/cargo-pgrx/default.nix (hunk 3 above)
Comment (kirillrdy) [reply pkgs/development/tools/rust/cargo-pgrx/default.nix:81]: This breaks vector-rs I'll have a look into that

Comment (Ma27) [reply pkgs/development/tools/rust/cargo-pgrx/default.nix:81]: Ah, bummer.
Nvm, you don't have to fix it up in this PR.

Comment (Ma27) [reply pkgs/development/tools/rust/cargo-pgrx/default.nix:81]: > Should we remove it?

Go for it.
I wouldn't do it on stable, but that's not the case here.

>  I wonder whether we should take responsibility for this with the @NixOS/postgres team as well?

Sounds good. Do you want to file a PR for that?

**Review by wolfgangwalther — APPROVED** [review]
Result of `nixpkgs-review pr 348223` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>postgresql13JitPackages.timescaledb</li>
    <li>postgresql13JitPackages.timescaledb-apache</li>
    <li>postgresql13Packages.timescaledb</li>
    <li>postgresql13Packages.timescaledb-apache</li>
  </ul>
</details>
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>27 packages built:</summary>
  <ul>
    <li>cargo-pgrx_0_12_6</li>
    <li>postgresql13JitPackages.timescaledb_toolkit</li>
    <li>postgresql13Packages.timescaledb_toolkit</li>
    <li>postgresql14JitPackages.timescaledb</li>
    <li>postgresql14JitPackages.timescaledb-apache</li>
    <li>postgresql14JitPackages.timescaledb_toolkit</li>
    <li>postgresql14Packages.timescaledb</li>
    <li>postgresql14Packages.timescaledb-apache</li>
    <li>postgresql14Packages.timescaledb_toolkit</li>
    <li>postgresql15JitPackages.timescaledb</li>
    <li>postgresql15JitPackages.timescaledb-apache</li>
    <li>postgresql15JitPackages.timescaledb_toolkit</li>
    <li>postgresql15Packages.timescaledb</li>
    <li>postgresql15Packages.timescaledb-apache</li>
    <li>postgresql15Packages.timescaledb_toolkit</li>
    <li>postgresql16JitPackages.timescaledb</li>
    <li>postgresql16JitPackages.timescaledb-apache</li>
    <li>postgresql16JitPackages.timescaledb_toolkit</li>
    <li>postgresql16Packages.timescaledb</li>
    <li>postgresql16Packages.timescaledb-apache</li>
    <li>postgresql16Packages.timescaledb_toolkit</li>
    <li>postgresql17JitPackages.timescaledb</li>
    <li>postgresql17JitPackages.timescaledb-apache</li>
    <li>postgresql17JitPackages.timescaledb_toolkit</li>
    <li>postgresql17Packages.timescaledb</li>
    <li>postgresql17Packages.timescaledb-apache</li>
    <li>postgresql17Packages.timescaledb_toolkit</li>
  </ul>
</details>

---

`nixosTests.postgresql.timescaledb` also built fine for all versions.

**Pull request merged**